core/lib/file-storage.js

```diff
@@ -0,0 +1,199 @@
+// Taken from https://github.com/VeliovGroup/Meteor-Files/blob/master/docs/aws-s3-integration.md
+import { Meteor } from 'meteor/meteor';
+import { _ } from 'meteor/underscore';
+import { Random } from 'meteor/random';
+import { FilesCollection } from 'meteor/ostrio:files';
+import stream from 'stream';
+
+import S3 from 'aws-sdk/clients/s3'; /* http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html */
+/* See fs-extra and graceful-fs NPM packages */
+/* For better i/o performance */
+import fs from 'fs';
+
+
+import { fileOnBeforeUpload } from './misc';
+
+function fileSystemAdapter() {
+  const s3Conf = Meteor.settings.s3;
+
+  if (!s3Conf) {
```

Comment:
A non-blocking detail, wouldn't it be better to put the empty S3 config in all settings files and check instead that the "s3?.key" attribute exists?

Reply:
Fixed in https://github.com/l3mpire/lemverse/commit/9299022b70527507c397099c35cf708ad9119a9b